[openmpi] `volumes/volumeMounts` support

Users sometimes wanted to add volumes to openmpi pods.  Particularly, when kubernetes cluster mounts NFS, users would want to mount `hostPath` volume to their containers.  Moreover, [kubernetes volumes](https://kubernetes.io/docs/concepts/storage/volumes/) supports various volume types.  So, this feature would also be useful for users who wanted to save their data to external volumes.

I experimented to implement this feature [on my branch](https://github.com/kubeflow/kubeflow/compare/master...everpeace:openmpi-volumes).  However,  this requires the latest ksonnet (0.10.2 or higher) to handle array of json parameter.  #727 blocks this issue.

/area openmpi
/priority p2

thank you @everpeace . That did help, able to mount volume now. :)

And now i'm trying to specify a ```  --nodeSelector=<nodeSelector>             Comma-delimited list of "key=value" pairs to select the worker nodes. e.g. "cloud.google.com/gke-accelerator=nvidia-tesla-k80" [default: null, type: string]
```
Above description is bit confusing, you say type as `string` but when explaining written as comma -delimited list. 

NODE_SELECTOR='["kubernetes.io/hostname=gke-x1", "kubernetes.io/hostname=gke-x2"]' - It didn't honor, randomly picked up some nodes.

or 
if it is string i tried as below
NODE_SELECTOR="kubernetes.io/hostname=gke-x1, gke-x2". It gave me below error
```level=error msg="find objects: RUNTIME ERROR: Index 1 out of bounds, not within [0, 1)\n\t/Users/kvadla/git-repo/mlt/examples/horovod_unet/mlt_app/unet-horovod-2e6fbba6-7e39-41f9-98b5-1f3241001e33/vendor/kubeflow/openmpi/util.libsonnet:14:46-81\tobject <anonymous>\n\t<std>:1222:30-34\tthunk from <thunk from <function <anonymous>>>\n\t<std>:1216:24-25\tthunk from <thunk <t> from <function <anonymous>>>\n\t<builtin>\tbuiltin function <type>\n\t<std>:1217:8-9\tfunction <anonymous>\n\t<std>:1165:25-26\tthunk from <thunk <ta> from <function <anonymous>>>\n\t<builtin>\tbuiltin function <type>\n\t<std>:1167:29-31\tthunk from <function <anonymous>>\n\t<builtin>\tbuiltin function <primitiveEquals>\n\t<std>:1167:8-36\tfunction <anonymous>\n\t\t\n\t<std>:1222:20-35\tthunk from <function <anonymous>>\n\t<std>:1208:10-11\tfunction <isContent>\n\t<std>:1165:25-26\tthunk from <thunk <ta> from <function <anonymous>>>\n\t<builtin>\tbuiltin function <type>\n\t<std>:1167:29-31\tthunk from <function <anonymous>>\n\t<builtin>\tbuiltin function <primitiveEquals>\n\t<std>:1167:8-36\tfunction <anonymous>\n\t\t\n\t<std>:1222:10-36\tfunction <anonymous>\n\t...\n\t<builtin>\tbuiltin function <type>\n\t<std>:1167:29-31\tthunk from <function <anonymous>>\n\t<builtin>\tbuiltin function <primitiveEquals>\n\t<std>:1167:8-36\tfunction <anonymous>\n\t\t\n\t<std>:1218:35-56\tfunction <anonymous>\n\t<builtin>\tbuiltin function <flatMap>\n\t<std>:1222:20-35\tthunk from <function <anonymous>>\n\t<std>:1208:10-11\tfunction <isContent>\n\t<std>:1165:25-26\tthunk from <thunk <ta> from <function <anonymous>>>\n\t<builtin>\tbuiltin function <type>\n\t<std>:1167:29-31\tthunk from <function <anonymous>>\n\t<builtin>\tbuiltin function <primitiveEquals>\n\t<std>:1167:8-36\tfunction <anonymous>\n\t\t\n\t<std>:1222:10-36\tfunction <anonymous>\n\t<builtin>\tbuiltin function <flatMap>\n\t<builtin>\tbuiltin function <$objectFlatMerge>\n\t<extvar:__ksonnet/components>:7:3-58\tobject <anonymous>\n\tDuring manifestation\t\n"
make: *** [deploy] Error 1
```

--- a/kubeflow/openmpi/prototypes/openmpi.jsonnet
+++ b/kubeflow/openmpi/prototypes/openmpi.jsonnet
@@ -28,6 +28,8 @@
 // @optionalParam runAsUser string null uid of the first process of containers in master/worker pods. If not set, this will be default value of your cluster configuration.
 // @optionalParam runAsGroup string null Primary gid of the first process of containers in master/worker pods. If not set, this will be default value of your cluster configuration.
 // @optionalParam supplementalGroups string null Comma-delimited list of supplemental group ids to put the user of the first process of containers in master/worker pods.
+// @optionalParam volumes array [] 'volumes' to put master/workers pods.
+// @optionalParam volumeMounts array [] 'volumes' to put job containers in master/workers pods.
 
 local k = import "k.libsonnet";
 local openmpi = import "kubeflow/openmpi/all.libsonnet";

--- a/kubeflow/openmpi/workloads.libsonnet
+++ b/kubeflow/openmpi/workloads.libsonnet
@@ -94,7 +94,7 @@ local ROLE_WORKER = "worker";
         defaultMode: 420,  // 0644
       },
     },
-  ],
+  ] + params.volumes,
 
   containers(params, role, podName):: {
     local job = {
@@ -132,7 +132,7 @@ local ROLE_WORKER = "worker";
           name: "openmpi-secrets",
           mountPath: "/kubeflow/openmpi/secrets",
         },
-      ],
+      ] + params.volumeMounts,
     },
     local controller = {
       name: "openmpi-controller",
@@ -148,7 +148,7 @@ local ROLE_WORKER = "worker";
           name: "openmpi-data",
           mountPath: "/kubeflow/openmpi/data",
         },
-      ],
+      ] + params.volumeMounts,
     },
 
     result:: if role == ROLE_MASTER then [job] else [job, controller],

